src/System.Net.Http/tests/FunctionalTests/DiagnosticsTests.cs

```diff
@@ -472,6 +472,10 @@ public void SendAsync_ExpectedDiagnosticExceptionActivityLogging()
         [Fact]
         public void SendAsync_ExpectedDiagnosticSynchronousExceptionActivityLogging()
         {
+            // For PlatformHandler (WinHttpHandler & CurlHandler), there is no easy way to force a
+            // synchronous exception during SendAsync.
+            if (!UseSocketsHttpHandler) return;
```

Comment:
PlatformHandler's SendAsync method is protected, and is invoked by HttpClient. As shown in source code, there is no easy way to force a synchronous exception, after the change for checking null value for CookieContainer.

WinHttpHandler:
https://github.com/dotnet/corefx/blob/master/src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs#L508-L534

CurlHandler:
https://github.com/dotnet/corefx/blob/master/src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.cs#L441-L471

I have tested that forcing an async Exception can also makes the test pass (by sending request to a invalid server). But this deviate from the purpose of this method: `SendAsync_ExpectedDiagnosticSynchronousExceptionActivityLogging`

If we think the coverage for PlatformHandler is important, I can add a new test method for it.

Reply:
We need to force sync exception from handler code (not from HttpClient), to make `diagnosticListenerObserver` to observe it.

> Won't passing a NULL HttpRequestMessage parameter generate that sync exception?

It will generate the exception, but from HttpClient. This will fail the test. Here is the process:

1. HttpClient/HttpMessageInvoker  will [intercept the Exception](https://github.com/dotnet/corefx/blob/master/src/System.Net.Http/src/System/Net/Http/HttpMessageInvoker.cs#L45-L51) before calling into handler code.
2. And the `HttpHandlerDiagnosticListener` in this test, will only observe the Exception from the HttpClientHandler.

There are other challenges to force Exception from handler code. We may could use Reflection to force sync exception during SendAsync in handler, but using Reflection doesn't fit the purpose for a functional test. Besides, if we want to verify HttpClientHandler's `SendAsync` behavior, there are many other tests in HttpClientHandlerTest.cs cover it.

Taken all these factors, I choose to skip the test for PlatformHandler.